src/main/java/io/zksync/domain/fee/TransactionType.java

```diff
@@ -22,4 +25,17 @@ public String getFeeIdentifier() {
         return feeIdentifier;
     }
 
+    public Object getRaw() {
+        if (this == TransactionType.CHANGE_PUB_KEY) {
+            ObjectMapper mapper = new ObjectMapper();
+            ObjectNode root = mapper.createObjectNode();
+            ObjectNode child = mapper.createObjectNode();
+            child.put("onchainPubkeyAuth", false);
```

Comment:
Why `onchainPubkeyAuth` is false? The users that will use `setAuthPubkeyHash` will work in the following flow:

1. The function `isOnChainAuthPubkeyHashSet` is called. If it's false then
`setAuthPubkeyHash` function is called.

2. After that, the user will call `setSigningKey` with onchainAuth equal to true. As far as I understand in the current implementation user can send such transactions, but there is no way to get the fee for it (the fees are different whether or not the `onchainPubkeyAuth` is true). 

If it is viable, I believe that the users should be able to choose whether or not `onchainPubkeyAuth` is true or false.

Reply:
Sure, that's just refactored legacy code. I'll add option for this tx type